fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,401 @@
-
-import math, os, bpy, bmesh, mathutils
-from bpy.utils import register_class, unregister_class
-from io import BytesIO
+import bpy
 
 from ..f3d.f3d_gbi import *
 from .oot_constants import *
 from .oot_utility import *
-
 from ..utility import *
+from .oot_operators import *
 
-class OOT_SearchActorIDEnumOperator(bpy.types.Operator):
-	bl_idname = "object.oot_search_actor_id_enum_operator"
-	bl_label = "Select Actor ID"
-	bl_property = "actorID"
-	bl_options = {'REGISTER', 'UNDO'} 
+# General classes and functions
 
-	actorID : bpy.props.EnumProperty(items = ootEnumActorID, default = "ACTOR_PLAYER")
-	actorUser : bpy.props.StringProperty(default = "Actor")
+class OOT_SearchChestContentEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_chest_content_enum_operator"
+	bl_label = "Select Chest Content"
+	bl_property = "itemChest"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	itemChest : bpy.props.EnumProperty(items = ootChestContent, default = '0x48')
 	objName : bpy.props.StringProperty()
 
 	def execute(self, context):
-		obj = bpy.data.objects[self.objName]
-		if self.actorUser == "Transition Actor":
-			obj.ootTransitionActorProperty.actor.actorID = self.actorID
-		elif self.actorUser == "Actor":
-			obj.ootActorProperty.actorID = self.actorID
-		elif self.actorUser == "Entrance":
-			obj.ootEntranceProperty.actor.actorID = self.actorID
-		else:
-			raise PluginError("Invalid actor user for search: " + str(self.actorUser))
+		bpy.data.objects[self.objName].fast64.oot.actor.itemChest = self.itemChest
+		bpy.context.region.tag_redraw()
+		self.report({'INFO'}, "Selected: " + self.itemChest)
+		return {'FINISHED'}
+
+	def invoke(self, context, event):
+		context.window_manager.invoke_search_popup(self)
+		return {'RUNNING_MODAL'}
+
+class OOT_SearchNaviMsgIDEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_navi_msg_id_enum_operator"
+	bl_label = "Select Message ID"
+	bl_property = "naviMsgID"
+	bl_options = {'REGISTER', 'UNDO'}
 
+	naviMsgID : bpy.props.EnumProperty(items = ootNaviMsgID, default = "0x00")
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		bpy.data.objects[self.objName].fast64.oot.actor.naviMsgID = self.naviMsgID
 		bpy.context.region.tag_redraw()
-		self.report({'INFO'}, "Selected: " + self.actorID)
+		self.report({'INFO'}, "Selected: " + self.naviMsgID)
 		return {'FINISHED'}
 
 	def invoke(self, context, event):
 		context.window_manager.invoke_search_popup(self)
 		return {'RUNNING_MODAL'}
 
+class OOTActorProperties(bpy.types.PropertyGroup):
+	# If you use the 'get=' option from Blender props don't actually save the data in the .blend
+	# When the get function is called we have to save the data that'll be returned
+	paramToSave = bpy.props.StringProperty()
+	transParamToSave = bpy.props.StringProperty()
+	XRotToSave = bpy.props.StringProperty(default='0x0')
+	YRotToSave = bpy.props.StringProperty(default='0x0')
+	ZRotToSave = bpy.props.StringProperty(default='0x0')
```

Comment:
you need to use `:` to define annotations
this is probably the issue you had when moving props to the class definition

https://docs.blender.org/api/current/bpy.props.html#propertygroup-example

Reply:
works fine lol
I just completely forgot you need to replace ``=`` by ``:`` when I first moved everything